go/indexer/services/l1/bridge/bridge.go

```diff
@@ -0,0 +1,88 @@
+package bridge
+
+import (
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/go/indexer/db"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type Bridge interface {
+	Address() common.Address
+	Contract() *bind.BoundContract
+	GetDepositsByBlockRange(uint64, uint64) (map[common.Hash][]db.Deposit, error)
+}
+
+var DEFAULT_L2_CONTRACT_ADDRESSES = map[string]string{
+	"OVM_L2ToL1MessagePasser": "0x4200000000000000000000000000000000000000",
+	"OVM_DeployerWhitelist":   "0x4200000000000000000000000000000000000002",
+	"L2CrossDomainMessenger":  "0x4200000000000000000000000000000000000007",
+	"OVM_GasPriceOracle":      "0x420000000000000000000000000000000000000F",
+	"L2StandardBridge":        "0x4200000000000000000000000000000000000010",
+	"OVM_SequencerFeeVault":   "0x4200000000000000000000000000000000000011",
+	"L2StandardTokenFactory":  "0x4200000000000000000000000000000000000012",
+	"OVM_L1BlockNumber":       "0x4200000000000000000000000000000000000013",
+
+	// We're temporarily disabling OVM_ETH because the jury is still out on whether or not ETH as an
+	// ERC20 is desirable.
+	"OVM_ETH": "0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000",
+
+	// We're also putting WETH9 at the old OVM_ETH address.
+	"WETH9": "0x4200000000000000000000000000000000000006",
+}
+
+var CONTRACT_ADDRESSES = map[uint]map[string]map[string]string{
```

Comment:
The L1 contracts can be fetched using the `AddressManager`, its less brittle than hardcoding them. The addressmanager addresses shouldn't change, except for hardhat local which could change if a new contract is added to the contracts package and then the order of the contracts being deployed is different. Goerli might change if we need to redeploy the system.

I think we should have a `--addressmanager 0x....` config that can override the hardcoded address manager and then only hardcode the address manager by L1 chain id. Then we can dynamically resolve the other contract addresses. What do you think?

Reply:
That can be done, and we can cache to result to not query the provider every time. However I see that the SDK among other things is also hardcoding these addresses. Do we have a standard policy around this across all repos?

https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/src/predeploys.tsS